File: frontend/src/app/model/ThreadDetailsDTO.ts
import { UserProfileDTO } from "./UserProfileDTO"

export interface ThreadDetailsDTO {
    encryptedId: string,
    title: string,
    content: string,
    createdAt: string,
    lastUpdatedAt: string,
    lastActivity: string,
    isClosed: boolean,
    user: UserProfileDTO,
    categoryName: string,
    subCategoryName: string,
    loggedUserThread: boolean
}

File: backend/dto/src/main/java/pl/dawid0604/pcForum/dto/thread/ThreadDetailsDTO.java
package pl.dawid0604.pcForum.dto.thread;

import pl.dawid0604.pcForum.dto.user.UserProfileDTO;

public record ThreadDetailsDTO(String encryptedId, String title, String content, String createdAt,
                               String lastUpdatedAt, String lastActivity, boolean isClosed,
                               UserProfileDTO user, String categoryName, String subCategoryName,
                               boolean loggedUserThread) { }
